JavaScript heap out of memory an OpenAPI 2.0 document

**Describe the bug**
I ran:

`spectral lint openapi.json`

which resulted in a stack trace. The input OpenAPI 2.0 file is attached as 
[openapi.json.txt](https://github.com/stoplightio/spectral/files/4766149/openapi.json.txt)
(The OpenAPI contains some unused elements because it was extracted from a more complete API definition.)

Stack trace attached as 

[spectral-traceback.txt](https://github.com/stoplightio/spectral/files/4766150/spectral-traceback.txt)

**Expected behavior**

complete normally.

**Environment (remove any that are not applicable):**
 - Library version: spectral 5.4.0
 - OS: mac osx 10.15.2
 - Browser: n/a (CLI)

**Additional context**

May be related to the size of the OpenAPI document. If I trim down some more of the content, eventually it works without running out of heap. But the source OpenAPI has many operations I left off of this, and other data.

Thanks a lot for all the details, really appreciate it.
I was able to narrow down the issue.
It's [json-to-ast](https://www.npmjs.com/package/json-to-ast) in [better-ajv-errors](https://github.com/atlassian/better-ajv-errors).
With better-ajv-errors disabled (I had to do it manually in the code and built the binary afterwards), I was able to lint the spec, with nimma enabled
```shell
USE_NIMMA=true /home/p0lip/Projects/Stoplight/spectral/binaries/spectral lint https://developer.apiture.com/docs/apis/accountApplications/v0.17.9/openapi.json
OpenAPI 2.0 (Swagger) detected

https://developer.apiture.com/docs/apis/accountApplications/v0.17.9/openapi.json
  1764:19    error  oas2-schema                    Invalid security definition.
  1768:27    error  oas2-schema                    should match format `uri`.
  2098:19    error  oas2-valid-definition-example  should match format `uri`
  2102:24    error  oas2-valid-definition-example  should match format `uri`
  2103:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2104:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2107:36    error  oas2-valid-definition-example  should match format `uri`
  2108:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2109:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2115:19    error  oas2-valid-definition-example  should match format `uri`
  2122:13    error  oas2-valid-definition-example  should match format `uri`
  2127:26    error  oas2-valid-definition-example  should match format `uri`
  2128:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2129:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2135:19    error  oas2-valid-definition-example  should match format `uri`
  2193:9     error  oas2-valid-parameter-example   should match format `uri`
  2199:9     error  oas2-valid-parameter-example   should be string
  2202:9     error  oas2-valid-parameter-example   should match format `uri`
  2207:18    error  oas2-valid-parameter-example   should match format `uri`
  2208:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2245:19    error  oas2-valid-parameter-example   should be string
  2267:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2275:19    error  oas2-valid-parameter-example   should match format `uri`
  2279:24    error  oas2-valid-parameter-example   should match format `uri`
  2280:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2281:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2284:36    error  oas2-valid-parameter-example   should match format `uri`
  2285:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2286:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2292:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2299:13    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2304:26    error  oas2-valid-parameter-example   should match format `uri`
  2305:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2306:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2312:19    error  oas2-valid-parameter-example   should match format `uri`
  2347:9     error  oas2-valid-definition-example  should match format `uri`
  2365:18    error  oas2-valid-definition-example  should match format `uri`
  2366:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2382:13    error  oas2-valid-definition-example  should match format `uri`
  2621:9     error  oas2-valid-parameter-example   should match format `uri`
  2624:9     error  oas2-valid-parameter-example   should be string
  2627:9     error  oas2-valid-parameter-example   should match format `uri`
  2644:18    error  oas2-valid-parameter-example   should match format `uri`
  2645:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2661:13    error  oas2-valid-parameter-example   should match format `uri`
  2762:19    error  oas2-valid-parameter-example   should be string
  2784:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2792:19    error  oas2-valid-parameter-example   should match format `uri`
  2796:24    error  oas2-valid-parameter-example   should match format `uri`
  2797:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2798:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2801:36    error  oas2-valid-parameter-example   should match format `uri`
  2802:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2803:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2809:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2816:13    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  2821:26    error  oas2-valid-parameter-example   should match format `uri`
  2822:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2823:19    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2829:19    error  oas2-valid-parameter-example   should match format `uri`
  2900:13    error  oas2-valid-definition-example  should be string
  2919:19    error  oas2-valid-definition-example  should be string
  2941:19    error  oas2-valid-definition-example  should match format `uri`
  2955:9     error  oas2-valid-parameter-example   should match format `uri`
  2958:9     error  oas2-valid-parameter-example   should be string
  2961:9     error  oas2-valid-parameter-example   should be string
  2966:18    error  oas2-valid-parameter-example   should match format `uri`
  2967:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  2982:13    error  oas2-valid-parameter-example   should be string
  2991:19    error  oas2-valid-parameter-example   should be string
  3013:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  3029:9     error  oas2-valid-definition-example  should be string
  3037:9     error  oas2-valid-definition-example  should match format `uri`
  3095:13    error  oas2-valid-definition-example  should be string
  3100:18    error  oas2-valid-definition-example  should match format `uri`
  3101:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3117:13    error  oas2-valid-definition-example  should match format `uri`
  3166:9     error  oas2-valid-parameter-example   should be string
  3169:9     error  oas2-valid-parameter-example   should be string
  3178:19    error  oas2-valid-parameter-example   should be string
  3200:19    error  oas2-valid-parameter-example   should be equal to one of the allowed values
  3257:13    error  oas2-valid-parameter-example   should be string
  3262:18    error  oas2-valid-parameter-example   should match format `uri`
  3263:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3279:13    error  oas2-valid-parameter-example   should match format `uri`
  3354:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3467:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3529:13    error  oas2-valid-definition-example  should be string
  3538:13    error  oas2-valid-definition-example  should be string
  3547:13    error  oas2-valid-definition-example  should match format `uri`
  3556:13    error  oas2-valid-definition-example  should match format `uri`
  3564:19    error  oas2-valid-definition-example  should match format `uri`
  3571:19    error  oas2-valid-definition-example  should match format `uri`
  3579:13    error  oas2-valid-definition-example  should match format `uri`
  3695:18    error  oas2-valid-definition-example  should match format `uri`
  3696:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3712:13    error  oas2-valid-definition-example  should match format `uri`
  3768:13    error  oas2-valid-definition-example  should match format `uri`
  3786:9     error  oas2-valid-definition-example  should match format `uri`
  3827:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  3864:21    error  oas2-valid-definition-example  should match format `uri`
  3941:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4039:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4101:17    error  oas2-valid-definition-example  should be object
  4128:23    error  oas2-valid-definition-example  should be object
  4128:23    error  oas2-valid-parameter-example   should be object
  4163:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4269:9     error  oas2-valid-definition-example  should match format `uri`
  4289:18    error  oas2-valid-definition-example  should match format `uri`
  4290:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4306:13    error  oas2-valid-definition-example  should match format `uri`
  4519:9     error  oas2-valid-definition-example  should match format `uri`
  4532:18    error  oas2-valid-definition-example  should match format `uri`
  4533:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4540:33    error  oas2-valid-definition-example  should match format `uri`
  4543:32    error  oas2-valid-definition-example  should match format `uri`
  4557:13    error  oas2-valid-definition-example  should match format `uri`
  4596:19    error  oas2-valid-definition-example  should match format `uri`
  4603:19    error  oas2-valid-definition-example  should match format `uri`
  4610:19    error  oas2-valid-definition-example  should match format `uri`
  4786:9     error  oas2-valid-definition-example  should match format `uri`
  4796:18    error  oas2-valid-definition-example  should match format `uri`
  4797:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4813:13    error  oas2-valid-definition-example  should match format `uri`
  4821:19    error  oas2-valid-definition-example  should match format `uri`
  4898:9     error  oas2-valid-definition-example  should match format `uri`
  4911:18    error  oas2-valid-definition-example  should match format `uri`
  4912:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  4918:13    error  oas2-valid-definition-example  should match format `uri`
  4932:13    error  oas2-valid-definition-example  should match format `uri`
  5007:9     error  oas2-valid-definition-example  should match format `uri`
  5022:18    error  oas2-valid-definition-example  should match format `uri`
  5023:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5039:13    error  oas2-valid-definition-example  should match format `uri`
  5100:18    error  oas2-valid-definition-example  should match format `uri`
  5101:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5118:21    error  oas2-valid-definition-example  should match format `uri`
  5142:9     error  oas2-valid-definition-example  should match format `uri`
  5152:18    error  oas2-valid-definition-example  should match format `uri`
  5153:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5169:13    error  oas2-valid-definition-example  should match format `uri`
  5185:9     error  oas2-valid-definition-example  should match format `uri`
  5198:18    error  oas2-valid-definition-example  should match format `uri`
  5199:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5215:13    error  oas2-valid-definition-example  should match format `uri`
  5350:9     error  oas2-valid-definition-example  should be string
  5404:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5458:19    error  oas2-valid-definition-example  should be string
  5513:18    error  oas2-valid-definition-example  should match format `uri`
  5514:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5530:13    error  oas2-valid-definition-example  should match format `uri`
  5580:9     error  oas2-valid-definition-example  should be string
  5583:9     error  oas2-valid-definition-example  should match format `uri`
  5603:19    error  oas2-valid-definition-example  should match format `uri`
  5604:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5619:19    error  oas2-valid-definition-example  should match format `uri`
  5663:19    error  oas2-valid-definition-example  should be string
  5758:19    error  oas2-valid-definition-example  should match format `uri`
  5763:18    error  oas2-valid-definition-example  should match format `uri`
  5764:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5780:13    error  oas2-valid-definition-example  should match format `uri`
  5822:9     error  oas2-valid-definition-example  should match format `uri`
  5852:18    error  oas2-valid-definition-example  should match format `uri`
  5853:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5869:13    error  oas2-valid-definition-example  should match format `uri`
  5925:9     error  oas2-valid-definition-example  should match format `uri`
  5935:18    error  oas2-valid-definition-example  should match format `uri`
  5936:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  5952:13    error  oas2-valid-definition-example  should match format `uri`
  5960:19    error  oas2-valid-definition-example  should match format `uri`
  6015:9     error  oas2-valid-definition-example  should match format `uri`
  6042:18    error  oas2-valid-definition-example  should match format `uri`
  6043:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  6059:13    error  oas2-valid-definition-example  should match format `uri`
  6174:9     error  oas2-valid-definition-example  should match format `uri`
  6188:18    error  oas2-valid-definition-example  should match format `uri`
  6189:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  6205:13    error  oas2-valid-definition-example  should match format `uri`
  6339:9     error  oas2-valid-definition-example  should match format `uri`
  6342:9     error  oas2-valid-definition-example  should match format `uri`
  6356:18    error  oas2-valid-definition-example  should match format `uri`
  6357:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  6373:13    error  oas2-valid-definition-example  should match format `uri`
  6391:17    error  oas2-valid-definition-example  should match format `uri`
  6441:9     error  oas2-valid-definition-example  should match format `uri`
  6453:18    error  oas2-valid-definition-example  should match format `uri`
  6454:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  6470:13    error  oas2-valid-definition-example  should match format `uri`
  6506:30    error  oas2-valid-definition-example  should match format `uri`
  6792:9     error  oas2-valid-definition-example  should match format `uri`
  6795:9     error  oas2-valid-definition-example  should match format `uri`
  6806:18    error  oas2-valid-definition-example  should match format `uri`
  6807:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  6823:13    error  oas2-valid-definition-example  should match format `uri`
  6862:19    error  oas2-valid-definition-example  should match format `uri`
  6869:19    error  oas2-valid-definition-example  should match format `uri`
  6876:19    error  oas2-valid-definition-example  should match format `uri`
  7038:9     error  oas2-valid-definition-example  should match format `uri`
  7048:18    error  oas2-valid-definition-example  should match format `uri`
  7049:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  7065:13    error  oas2-valid-definition-example  should match format `uri`
  7124:9     error  oas2-valid-definition-example  should match format `uri`
  7134:18    error  oas2-valid-definition-example  should match format `uri`
  7135:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  7151:13    error  oas2-valid-definition-example  should match format `uri`
  7160:19    error  oas2-valid-definition-example  should be string
  7179:9     error  oas2-valid-definition-example  should match format `uri`
  7189:18    error  oas2-valid-definition-example  should match format `uri`
  7190:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  7206:13    error  oas2-valid-definition-example  should match format `uri`
  7267:13    error  oas2-valid-definition-example  should match format `uri`
  7369:18    error  oas2-valid-definition-example  should match format `uri`
  7432:18    error  oas2-valid-definition-example  should match format `uri`
  7543:9     error  oas2-valid-definition-example  should be string
  7549:9     error  oas2-valid-definition-example  should be string
  7584:19    error  oas2-valid-definition-example  should match format `uri`
  7585:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  7600:19    error  oas2-valid-definition-example  should match format `uri`
  7644:19    error  oas2-valid-definition-example  should be string
  7739:19    error  oas2-valid-definition-example  should match format `uri`
  7855:9     error  oas2-valid-definition-example  should be string
  7858:9     error  oas2-valid-definition-example  should match format `uri`
  7911:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  7965:19    error  oas2-valid-definition-example  should be string
  8020:18    error  oas2-valid-definition-example  should match format `uri`
  8021:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8037:13    error  oas2-valid-definition-example  should match format `uri`
  8069:18    error  oas2-valid-definition-example  should match format `uri`
  8095:21    error  oas2-valid-definition-example  should match format `uri`
  8118:9     error  oas2-valid-definition-example  should match format `uri`
  8143:18    error  oas2-valid-definition-example  should match format `uri`
  8144:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8160:13    error  oas2-valid-definition-example  should match format `uri`
  8211:9     error  oas2-valid-definition-example  should match format `uri`
  8238:18    error  oas2-valid-definition-example  should match format `uri`
  8239:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8255:13    error  oas2-valid-definition-example  should match format `uri`
  8415:9     error  oas2-valid-definition-example  should match format `uri`
  8426:18    error  oas2-valid-definition-example  should match format `uri`
  8427:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8443:13    error  oas2-valid-definition-example  should match format `uri`
  8504:9     error  oas2-valid-definition-example  should match format `uri`
  8541:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8553:18    error  oas2-valid-definition-example  should match format `uri`
  8554:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  8658:17    error  oas2-valid-definition-example  should match format `uri`
  8744:9     error  oas2-valid-definition-example  should match format `uri`
  8806:19    error  oas2-valid-definition-example  should match format `uri`
  8813:19    error  oas2-valid-definition-example  should match format `uri`
  8820:19    error  oas2-valid-definition-example  should match format `uri`
  9109:21    error  oas2-valid-definition-example  should be string
  9117:9     error  oas2-valid-definition-example  should be string
  9146:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  9200:19    error  oas2-valid-definition-example  should be string
  9252:9     error  oas2-valid-definition-example  should match format `uri`
  9255:9     error  oas2-valid-definition-example  should be equal to one of the allowed values
  9265:18    error  oas2-valid-definition-example  should match format `uri`
  9266:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  9282:13    error  oas2-valid-definition-example  should match format `uri`
  9413:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  9481:9     error  oas2-valid-definition-example  should match format `uri`
  9494:18    error  oas2-valid-definition-example  should match format `uri`
  9495:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  9511:13    error  oas2-valid-definition-example  should match format `uri`
  9684:9     error  oas2-valid-definition-example  should be string
  9708:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  9762:19    error  oas2-valid-definition-example  should be string
  9890:19    error  oas2-valid-definition-example  should be string
  9980:9     error  oas2-valid-definition-example  should match format `uri`
 10017:18    error  oas2-valid-definition-example  should match format `uri`
 10018:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 10034:13    error  oas2-valid-definition-example  should match format `uri`
  10088:9    error  oas2-valid-definition-example  should match format `uri`
 10122:18    error  oas2-valid-definition-example  should match format `uri`
 10123:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 10139:13    error  oas2-valid-definition-example  should match format `uri`
  10199:9    error  oas2-valid-definition-example  should match format `uri`
 10228:18    error  oas2-valid-definition-example  should match format `uri`
 10229:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 10245:13    error  oas2-valid-definition-example  should match format `uri`
  10299:9    error  oas2-valid-definition-example  should match format `uri`
 10329:18    error  oas2-valid-definition-example  should match format `uri`
 10330:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 10346:13    error  oas2-valid-definition-example  should match format `uri`
 10647:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 10868:19    error  oas2-valid-definition-example  should match format `uri`
 10875:19    error  oas2-valid-definition-example  should match format `uri`
 10882:19    error  oas2-valid-definition-example  should match format `uri`
 11158:18    error  oas2-valid-definition-example  should be equal to one of the allowed values
 11211:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
  11325:9    error  oas2-valid-definition-example  should be string
 11346:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 11400:19    error  oas2-valid-definition-example  should be string
 11672:22    error  oas2-valid-definition-example  should be string
  12035:9    error  oas2-valid-definition-example  should match format `uri`
 12072:18    error  oas2-valid-definition-example  should match format `uri`
 12073:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 12089:13    error  oas2-valid-definition-example  should match format `uri`
  12143:9    error  oas2-valid-definition-example  should match format `uri`
 12177:18    error  oas2-valid-definition-example  should match format `uri`
 12178:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 12194:13    error  oas2-valid-definition-example  should match format `uri`
  12254:9    error  oas2-valid-definition-example  should match format `uri`
 12283:18    error  oas2-valid-definition-example  should match format `uri`
 12284:26    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 12300:13    error  oas2-valid-definition-example  should match format `uri`
 12410:21    error  no-$ref-siblings               $ref cannot be placed next to any other properties
 12683:19  warning  oas2-unused-definition         Potentially unused definition has been detected.

✖ 308 problems (307 errors, 1 warning, 0 infos, 0 hints)

```

I'm going to fork better-ajv-errors and get rid of json-to-ast entirely, as we don't need it.

--- a/package.json
+++ b/package.json
@@ -60,6 +60,7 @@
     "test": "jest --silent"
   },
   "dependencies": {
+    "@stoplight/better-ajv-errors": "https://github.com/stoplightio/better-ajv-errors.git#bundled",
     "@stoplight/json": "3.6.0",
     "@stoplight/json-ref-readers": "1.1.1",
     "@stoplight/json-ref-resolver": "3.0.9",
@@ -70,7 +71,6 @@
     "abort-controller": "3.0.0",
     "ajv": "6.12.2",
     "ajv-oai": "1.2.0",
-    "better-ajv-errors": "0.6.7",
     "blueimp-md5": "2.13.0",
     "chalk": "4.0.0",
     "eol": "0.9.1",
@@ -83,7 +83,7 @@
     "proxy-agent": "3.1.1",
     "strip-ansi": "6.0",
     "text-table": "0.2",
-    "tslib": "1.11.1",
+    "tslib": "1.13.0",
     "yargs": "15.3.1"
   },
   "devDependencies": {

--- a/src/cli/services/__tests__/linter.test.ts
+++ b/src/cli/services/__tests__/linter.test.ts
@@ -651,7 +651,7 @@ describe('Linter service', () => {
         }),
         expect.objectContaining({
           code: 'oas2-schema',
-          message: `Object should have required property \`title\`.`,
+          message: '`info` property should have required property `title`.',
           path: [],
           range: {
             end: {

--- a/src/functions/__tests__/schema-path.test.ts
+++ b/src/functions/__tests__/schema-path.test.ts
@@ -64,7 +64,7 @@ describe('schema-path', () => {
     expect(runSchemaPath(target, fieldToCheck, path)).toEqual([
       {
         path: ['example'],
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
       },
     ]);
   });
@@ -79,7 +79,7 @@ describe('schema-path', () => {
     expect(runSchemaPath(target, fieldToCheck, path)).toEqual([
       {
         path: ['example'],
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be string',
+        message: '`example` property type should be string',
       },
     ]);
   });
@@ -109,11 +109,11 @@ describe('schema-path', () => {
     };
     expect(runSchemaPath(target, '$.examples.*', path)).toEqual([
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be string',
+        message: '`id` property type should be string',
         path: ['examples', 'application/json', 'id'],
       },
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be string',
+        message: '`id` property type should be string',
         path: ['examples', 'application/yaml', 'id'],
       },
     ]);
@@ -138,11 +138,11 @@ describe('schema-path', () => {
     };
     expect(runSchemaPath(target, '$.schema.examples.*', path)).toEqual([
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be string',
+        message: '`0` property type should be string',
         path: ['schema', 'examples', '0'],
       },
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be string',
+        message: '`2` property type should be string',
         path: ['schema', 'examples', '2'],
       },
     ]);
@@ -159,7 +159,7 @@ describe('schema-path', () => {
 
     expect(runSchemaPath(target, fieldToCheck, path)).toEqual([
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}format should match format `url`',
+        message: '`example` property should match format `url`',
         path: ['example'],
       },
     ]);

--- a/src/functions/__tests__/schema.test.ts
+++ b/src/functions/__tests__/schema.test.ts
@@ -25,7 +25,7 @@ describe('schema', () => {
 
       expect(runSchema('', testSchema)).toEqual([
         {
-          message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be number',
+          message: 'Value type should be number',
           path: [],
         },
       ]);
@@ -38,7 +38,7 @@ describe('schema', () => {
 
       expect(runSchema(0, testSchema)).toEqual([
         {
-          message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+          message: `Value type should be string`,
           path: [],
         },
       ]);
@@ -51,7 +51,7 @@ describe('schema', () => {
 
       expect(runSchema(false, testSchema)).toEqual([
         {
-          message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+          message: `Value type should be string`,
           path: [],
         },
       ]);
@@ -64,7 +64,7 @@ describe('schema', () => {
 
       expect(runSchema(null, testSchema)).toEqual([
         {
-          message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+          message: `Value type should be string`,
           path: [],
         },
       ]);
@@ -105,7 +105,7 @@ describe('schema', () => {
       const input = { foo: 'bar' };
       expect(runSchema(input, testSchema)).toEqual([
         expect.objectContaining({
-          message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be array',
+          message: 'Value type should be array',
           path: [],
         }),
       ]);
@@ -115,8 +115,7 @@ describe('schema', () => {
       const input = ['1', '2'];
       expect(runSchema(input, testSchema)).toEqual([
         expect.objectContaining({
-          message:
-            '{{property|gravis|append-property|optional-typeof|capitalize}}maxItems should NOT have more than 1 items',
+          message: 'Object should not have more than 1 items',
           path: [],
         }),
       ]);
@@ -152,7 +151,7 @@ describe('schema', () => {
         ),
       ).toEqual([
         {
-          message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+          message: '`bar` property type should be string',
           path: ['foo', 'bar'],
         },
       ]);
@@ -186,7 +185,7 @@ describe('schema', () => {
       const input = 'not an email';
       expect(runSchema(input, testSchema)).toEqual([
         expect.objectContaining({
-          message: '{{property|gravis|append-property|optional-typeof|capitalize}}format should match format `email`',
+          message: 'String should match format `email`',
           path: [],
         }),
       ]);
@@ -224,7 +223,7 @@ describe('schema', () => {
     expect(runSchema(2, testSchema)).toEqual([
       {
         path: [],
-        message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+        message: `Value type should be string`,
       },
     ]);
     expect(runSchema('a', testSchema2)).toEqual([]);
@@ -244,7 +243,7 @@ describe('schema', () => {
     expect(runSchema(2, testSchema)).toEqual([
       {
         path: [],
-        message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+        message: `Value type should be string`,
       },
     ]);
     expect(runSchema('a', testSchema2)).toEqual([]);
@@ -279,7 +278,7 @@ describe('schema', () => {
       it('reports pretty enum errors for a number', () => {
         expect(runSchema(2, testSchema)).toEqual([
           {
-            message: `{{property|gravis|append-property|optional-typeof|capitalize}}type should be string`,
+            message: 'Value type should be string',
             path: [],
           },
         ]);
@@ -296,7 +295,7 @@ describe('schema', () => {
       it('reports pretty enum errors for a string', () => {
         expect(runSchema('baz', testSchema)).toEqual([
           {
-            message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be integer',
+            message: 'Value type should be integer',
             path: [],
           },
         ]);

--- a/src/functions/schema.ts
+++ b/src/functions/schema.ts
@@ -1,15 +1,12 @@
-import { decodePointerFragment } from '@stoplight/json';
 import { Optional } from '@stoplight/types';
 import * as AJV from 'ajv';
 import { ValidateFunction } from 'ajv';
 import * as jsonSpecV4 from 'ajv/lib/refs/json-schema-draft-04.json';
 import * as jsonSpecV6 from 'ajv/lib/refs/json-schema-draft-06.json';
 import * as jsonSpecV7 from 'ajv/lib/refs/json-schema-draft-07.json';
-import { IOutputError } from 'better-ajv-errors';
-import { capitalize, escapeRegExp } from 'lodash';
+import * as betterAjvErrors from '@stoplight/better-ajv-errors';
 import { IFunction, IFunctionResult, JSONSchema } from '../types';
 const oasFormatValidator = require('ajv-oai/lib/format-validator');
-const betterAjvErrors = require('better-ajv-errors/lib/modern');
 
 export interface ISchemaFunction extends IFunction<ISchemaOptions> {
   Ajv: typeof AJV;
@@ -32,12 +29,8 @@ export interface ISchemaOptions {
   prepareResults?(errors: AJV.ErrorObject[]): void;
 }
 
-interface IAJVOutputError extends IOutputError {
-  path?: string;
-}
-
 const logger = {
-  warn(...args: any[]) {
+  warn(...args: unknown[]): void {
     const firstArg = args[0];
     if (typeof firstArg === 'string') {
       if (firstArg.startsWith('unknown format')) return;
@@ -51,7 +44,7 @@ const logger = {
 const ajvInstances = {};
 
 function getAjv(oasVersion: Optional<number>, allErrors: Optional<boolean>): AJV.Ajv {
-  const type: string = oasVersion && oasVersion >= 2 ? 'oas' + oasVersion : 'jsonschema';
+  const type: string = oasVersion !== void 0 && oasVersion >= 2 ? 'oas' + oasVersion : 'jsonschema';
   if (typeof ajvInstances[type] !== 'undefined') {
     return ajvInstances[type];
   }
@@ -92,7 +85,7 @@ function getSchemaId(schemaObj: JSONSchema): void | string {
 }
 
 const validators = new (class extends WeakMap<JSONSchema, ValidateFunction> {
-  public get({ schema: schemaObj, oasVersion, allErrors }: ISchemaOptions) {
+  public get({ schema: schemaObj, oasVersion, allErrors }: ISchemaOptions): ValidateFunction {
     const ajv = getAjv(oasVersion, allErrors);
     const schemaId = getSchemaId(schemaObj);
     let validator = schemaId !== void 0 ? ajv.getSchema(schemaId) : void 0;
@@ -110,39 +103,6 @@ const validators = new (class extends WeakMap<JSONSchema, ValidateFunction> {
     return validator;
   }
 })();
-
-const replaceProperty = (
-  substring: string,
-  potentialProperty: Optional<number | string>,
-  propertyName: Optional<string>,
-) => {
-  if (typeof potentialProperty === 'string' && propertyName !== void 0) {
-    return `Property \`${propertyName}\``;
-  }
-
-  return '{{property|gravis|append-property|optional-typeof|capitalize}}';
-};
-
-const cleanAJVErrorMessage = (message: string, path: Optional<string>, suggestion: Optional<string>, type: string) => {
-  let cleanMessage = message.trim();
-
-  if (path) {
-    cleanMessage = message.replace(
-      new RegExp(`^${escapeRegExp(decodePointerFragment(path))}:?\\s*(?:(Property\\s+)([^\\s]+))?`),
-      replaceProperty,
-    );
-  } else if (cleanMessage.startsWith(':')) {
-    cleanMessage = cleanMessage.replace(/:\s*/, replaceProperty);
-  } else if (cleanMessage.startsWith('Property ')) {
-    cleanMessage = cleanMessage.replace(/(Property\s+)([^\s]+)/, replaceProperty);
-  } else {
-    cleanMessage = `${capitalize(type)} ${cleanMessage}`;
-  }
-
-  return `${cleanMessage.replace(/['"]/g, '`')}${
-    typeof suggestion === 'string' && suggestion.length > 0 ? `. ${suggestion}` : ''
-  }`;
-};
 
 export const schema: ISchemaFunction = (targetVal, opts, paths, { rule }) => {
   const path = paths.target ?? paths.given;
@@ -164,26 +124,18 @@ export const schema: ISchemaFunction = (targetVal, opts, paths, { rule }) => {
   try {
     // we used the compiled validation now, hence this lookup here (see the logic above for more info)
     const validator = opts.ajv ?? validators.get(opts);
-    if (!validator(targetVal) && validator.errors) {
+    if (validator(targetVal) === false && Array.isArray(validator.errors)) {
       opts.prepareResults?.(validator.errors);
 
-      try {
-        results.push(
-          ...(betterAjvErrors(schemaObj, targetVal, validator.errors, { format: 'js' }) as IAJVOutputError[]).map(
-            ({ suggestion, error, path: errorPath }) => ({
-              message: cleanAJVErrorMessage(error, errorPath, suggestion, typeof targetVal),
-              path: [...path, ...(errorPath ? errorPath.replace(/^\//, '').split('/') : [])],
-            }),
-          ),
-        );
-      } catch {
-        results.push(
-          ...validator.errors.map(({ message, dataPath }) => ({
-            message: message ? cleanAJVErrorMessage(message, dataPath, void 0, typeof targetVal) : '',
-            path: [...path, ...dataPath.split('/').slice(1).map(decodePointerFragment)],
-          })),
-        );
-      }
+      results.push(
+        ...betterAjvErrors(schemaObj, validator.errors, {
+          propertyPath: path,
+          targetValue: targetVal,
+        }).map(({ suggestion, error, path: errorPath }) => ({
+          message: suggestion !== void 0 ? `${error}. ${suggestion}` : error,
+          path: [...path, ...(errorPath !== '' ? errorPath.replace(/^\//, '').split('/') : [])],
+        })),
+      );
     }
   } catch (ex) {
     if (!(ex instanceof AJV.MissingRefError)) {
@@ -204,7 +156,7 @@ export const schema: ISchemaFunction = (targetVal, opts, paths, { rule }) => {
 
 schema.Ajv = AJV;
 // eslint-disable-next-line @typescript-eslint/unbound-method
-schema.createAJVInstance = (opts: AJV.Options) => {
+schema.createAJVInstance = (opts: AJV.Options): AJV.Ajv => {
   const ajv = new AJV(opts);
 
   ajv.addFormat('int32', { type: 'number', validate: oasFormatValidator.int32 });

--- a/src/rulesets/asyncapi/__tests__/asyncapi-headers-schema-type-object.ts
+++ b/src/rulesets/asyncapi/__tests__/asyncapi-headers-schema-type-object.ts
@@ -80,7 +80,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Headers schema type should be `object` (Object should have required property `type`).',
+        message: 'Headers schema type should be `object` (`headers` property should have required property `type`).',
         path: ['components', 'messages', 'aMessage', 'headers'],
         severity: rule.severity,
       }),
@@ -115,7 +115,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Headers schema type should be `object` (Object should have required property `type`).',
+        message: 'Headers schema type should be `object` (`headers` property should have required property `type`).',
         path: ['components', 'messageTraits', 'aTrait', 'headers'],
         severity: rule.severity,
       }),
@@ -134,7 +134,7 @@ describe(`Rule '${ruleName}'`, () => {
       expect(results).toEqual([
         expect.objectContaining({
           code: ruleName,
-          message: 'Headers schema type should be `object` (Object should have required property `type`).',
+          message: 'Headers schema type should be `object` (`headers` property should have required property `type`).',
           path: ['channels', 'users/{userId}/signedUp', property, 'message', 'headers'],
           severity: rule.severity,
         }),

--- a/src/rulesets/asyncapi/__tests__/asyncapi-payload-default.ts
+++ b/src/rulesets/asyncapi/__tests__/asyncapi-payload-default.ts
@@ -70,7 +70,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `value`',
+        message: '`default` property should have required property `value`',
         path: ['components', 'messages', 'aMessage', 'payload', 'default'],
         severity: rule.severity,
       }),
@@ -87,7 +87,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `value`',
+        message: '`default` property should have required property `value`',
         path: ['components', 'messageTraits', 'aTrait', 'payload', 'default'],
         severity: rule.severity,
       }),
@@ -106,7 +106,7 @@ describe(`Rule '${ruleName}'`, () => {
       expect(results).toEqual([
         expect.objectContaining({
           code: ruleName,
-          message: 'Object should have required property `value`',
+          message: '`default` property should have required property `value`',
           path: ['channels', 'users/{userId}/signedUp', property, 'message', 'payload', 'default'],
           severity: rule.severity,
         }),

--- a/src/rulesets/asyncapi/__tests__/asyncapi-payload-examples.ts
+++ b/src/rulesets/asyncapi/__tests__/asyncapi-payload-examples.ts
@@ -70,7 +70,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `value`',
+        message: '`1` property should have required property `value`',
         path: ['components', 'messages', 'aMessage', 'payload', 'examples', '1'],
         severity: rule.severity,
       }),
@@ -87,7 +87,7 @@ describe(`Rule '${ruleName}'`, () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `value`',
+        message: '`1` property should have required property `value`',
         path: ['components', 'messageTraits', 'aTrait', 'payload', 'examples', '1'],
         severity: rule.severity,
       }),
@@ -106,7 +106,7 @@ describe(`Rule '${ruleName}'`, () => {
       expect(results).toEqual([
         expect.objectContaining({
           code: ruleName,
-          message: 'Object should have required property `value`',
+          message: '`1` property should have required property `value`',
           path: ['channels', 'users/{userId}/signedUp', property, 'message', 'payload', 'examples', '1'],
           severity: rule.severity,
         }),

--- a/src/rulesets/asyncapi/functions/__tests__/asyncApi2PayloadValidation.test.ts
+++ b/src/rulesets/asyncapi/functions/__tests__/asyncApi2PayloadValidation.test.ts
@@ -22,7 +22,7 @@ describe('asyncApi2PayloadValidation', () => {
 
     expect(results).toEqual([
       {
-        message: '{{property|gravis|append-property|optional-typeof|capitalize}}type should be boolean',
+        message: '`deprecated` property type should be boolean',
         path: ['$', 'components', 'messages', 'aMessage', 'deprecated'],
       },
     ]);

--- a/src/rulesets/oas/__tests__/oas2-valid-definition-example.ts
+++ b/src/rulesets/oas/__tests__/oas2-valid-definition-example.ts
@@ -108,7 +108,7 @@ describe('oas2-valid-definition-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas2-valid-definition-example',
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);

--- a/src/rulesets/oas/__tests__/oas2-valid-parameter-example.ts
+++ b/src/rulesets/oas/__tests__/oas2-valid-parameter-example.ts
@@ -131,7 +131,7 @@ describe('oas2-valid-parameter-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas2-valid-parameter-example',
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);

--- a/src/rulesets/oas/__tests__/oas3-valid-oas-parameter-example.ts
+++ b/src/rulesets/oas/__tests__/oas3-valid-oas-parameter-example.ts
@@ -124,7 +124,7 @@ describe('oas3-valid-oas-parameter-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas3-valid-oas-parameter-example',
-        message: 'Object should have required property `abc`',
+        message: '`example` property should have required property `abc`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);
@@ -161,7 +161,7 @@ describe('oas3-valid-oas-parameter-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas3-valid-oas-parameter-example',
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);

--- a/src/rulesets/oas/__tests__/oas3-valid-parameter-schema-example.ts
+++ b/src/rulesets/oas/__tests__/oas3-valid-parameter-schema-example.ts
@@ -288,7 +288,7 @@ describe('oas3-valid-parameter-schema-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas3-valid-parameter-schema-example',
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);

--- a/src/rulesets/oas/__tests__/oas3-valid-schema-example.ts
+++ b/src/rulesets/oas/__tests__/oas3-valid-schema-example.ts
@@ -230,7 +230,7 @@ describe('oas3-valid-schema-example', () => {
     expect(results).toEqual([
       expect.objectContaining({
         code: 'oas3-valid-schema-example',
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);

--- a/src/rulesets/oas/__tests__/templates/_oas-example.ts
+++ b/src/rulesets/oas/__tests__/templates/_oas-example.ts
@@ -162,7 +162,7 @@ export default (ruleName: string, path: string) => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);
@@ -213,7 +213,7 @@ export default (ruleName: string, path: string) => {
       expect.objectContaining({
         severity: DiagnosticSeverity.Error,
         code: ruleName,
-        message: `\`example\` property format should match format \`${format}\``,
+        message: `\`example\` property should match format \`${format}\``,
       }),
     ]);
   });

--- a/src/rulesets/oas/__tests__/templates/_schema-example.ts
+++ b/src/rulesets/oas/__tests__/templates/_schema-example.ts
@@ -216,7 +216,7 @@ export default (ruleName: string, path: string) => {
     expect(results).toEqual([
       expect.objectContaining({
         code: ruleName,
-        message: 'Object should have required property `url`',
+        message: '`example` property should have required property `url`',
         severity: DiagnosticSeverity.Error,
       }),
     ]);
@@ -272,7 +272,7 @@ export default (ruleName: string, path: string) => {
       expect.objectContaining({
         severity: DiagnosticSeverity.Error,
         code: ruleName,
-        message: `\`example\` property format should match format \`${format}\``,
+        message: `\`example\` property should match format \`${format}\``,
       }),
     ]);
   });

--- a/test-harness/scenarios/external-schemas-ruleset.scenario
+++ b/test-harness/scenarios/external-schemas-ruleset.scenario
@@ -11,7 +11,7 @@ info:
 OpenAPI 3.x detected
 
 {document}
- 3:10  error  info-title        String should be equal to one of the allowed values: Stoplight, Stoplight.io, StoplightIO. Did you mean Stoplight?
- 4:16  error  info-description  String should be equal to one of the allowed values: foo, foo-bar, bar-foo
+ 3:10  error  info-title        `title` property should be equal to one of the allowed values: Stoplight, Stoplight.io, StoplightIO. Did you mean Stoplight?
+ 4:16  error  info-description  `description` property should be equal to one of the allowed values: foo, foo-bar, bar-foo
 
 ✖ 2 problems (2 errors, 0 warnings, 0 infos, 0 hints)

--- a/test-harness/scenarios/oas3-schema.scenario
+++ b/test-harness/scenarios/oas3-schema.scenario
@@ -58,6 +58,6 @@ OpenAPI 3.x detected
  32:28  error  oas3-schema  `user_id` property type should be object.
  36:28  error  oas3-schema  `properties` property type should be object.
  38:23  error  oas3-schema  `description` property type should be string.
- 41:17  error  oas3-schema  `responses` property minProperties should NOT have fewer than 1 properties.
+ 41:17  error  oas3-schema  `responses` property should not have fewer than 1 properties.
 
 ✖ 7 problems (7 errors, 0 warnings, 0 infos, 0 hints)

